src/main/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureService.scala

```diff
@@ -50,23 +93,25 @@ class AzureService(crlService: CrlService, directoryDAO: DirectoryDAO) {
   private def createUserPetManagedIdentity(
       id: PetManagedIdentityId,
       user: SamUser,
-      request: GetOrCreatePetManagedIdentityRequest,
+      mrgCoords: ManagedResourceGroupCoordinates,
       samRequestContext: SamRequestContext
   ): IO[(PetManagedIdentity, Boolean)] =
     for {
-      _ <- validateManagedResourceGroup(request)
```

Comment:
this validate call was moved to `getBillingProfileIdFromAzureTag` for 2 reasons, 
1. it only needs to be called when getting the billing profile id from a tag because we need to verify we trust the MRG, this check is not required when getting the billing profile id from the db because it is done before saving it
2. checking here is a little late in the game since we have already read and used the tag before verifying trust.